### PR TITLE
Feature/8 button

### PR DESCRIPTION
**Pull request by samumic**

### CODE REVIEW

File: src/components/button/button.tsx [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
-  children: ReactNode;
-};
+    if(props.size === ButtonSize.M) base.push('p-s');
+    if(props.size === ButtonSize.L) base.push('py-m px-s');
 
-export const Button: FC<ButtonProps> = ({ type, children }) => (
-  <button className={type === ButtonType.primary ? 'bg-blue-500 text-white' : 'bg-blue-200 text-black'}>{children}</button>
-);
+    return (
+        <button className={props.onlyIcon ? icon : base.join(' ')}>
```
Comment (monobasic) [inline src/components/button/button.tsx:16]: I would also add aria-label here as a minimum accessibility feature

File: src/components/button/button.tsx [hunk 2]
```diff
@@ -1,15 +1,25 @@
-import React, { FC, ReactNode } from 'react';
+import React from 'react';
+import {ButtonProps, ButtonSize, ButtonType} from "./button.types";
 
-export enum ButtonType {
-  primary = 'primary',
-  secondary = 'secondary',
-}
+export const Button: React.FC<ButtonProps> = (props) => {
+    const base = ['flex items-center text-white rounded-default border-4 border-white'];
```
Comment (monobasic) [inline src/components/button/button.tsx:5]: Let's use descriptive variable names, as example: baseStyles or baseClasses

File: src/components/button/button.tsx [hunk 3]
```diff
@@ -1,15 +1,25 @@
...
+    const icon = 'border-4 border-white text-white rounded-38 p-m bg-slate-600 hover:bg-slate-700 hover:border-slate-100 focus:border-slate-200';
```
Comment (monobasic) [inline src/components/button/button.tsx:6]: Same here.. iconClasses or something..

File: src/components/button/button.types.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    M = 'm',
+    L = 'l'
+}
+
+export type ButtonProps = {
+    type?: ButtonType;
+    size?: ButtonSize;
+    label?: ReactNode | string;
+    children?: ReactNode;
+    onlyIcon?: boolean;
```
Comment (monobasic) [inline src/components/button/button.types.ts:19]: I would go with is.. or should.. for Booleans. example: isIconOnly or shouldRenderIconOnly
see: https://github.com/kettanaito/naming-cheatsheet

File: tailwind.config.js [hunk 5]
```diff
@@ -89,6 +89,282 @@ module.exports = {
       semibold: "600",
       bold: "700",
     },
+    borderRadius: {
```
Comment (monobasic) [inline tailwind.config.js:92]: Woah.. needed to add all the configs back? Please explain me on Tuesday..

File: src/components/components.config.js [hunk 6]
```diff
@@ -0,0 +1,3 @@
+export const cosyMeta = {
```
Comment (monobasic) [inline src/components/components.config.js:1]: I have no idea, what this is :D

File: .storybook/main.js [hunk 7]
```diff
@@ -2,62 +2,62 @@ const path = require("path");
...
+                postcssLoaderOptions: {
+                    implementation: require("postcss"),
+                },
+            },
         },
-        postcssLoaderOptions: {
-          implementation: require("postcss"),
+    ],
+    framework: "@storybook/react",
+    typescript: {
```
Comment (monobasic) [inline .storybook/main.js:23]: Need your explanation here on Tuesday :)

File: src/components/components.config.js (hunk 6 above)
Comment (samumic) [reply src/components/components.config.js:1]: Let's rename it to -> StorybookMeta

File: tailwind.config.js (hunk 5 above)
Comment (samumic) [reply tailwind.config.js:92]: We should move our styles to extend object

Comment (samumic) [reply tailwind.config.js:92]: Actually, Presets:[], was the issue that our classes werent, generated, thanks for the input

File: src/components/button/button.types.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+export enum ButtonSize {
+    M = 'm',
+    L = 'l'
+}
+
+export type ButtonProps = {
+    type?: ButtonType;
+    size?: ButtonSize;
+    label?: ReactNode | string;
```
[the remark this answers is not in the document]
Comment (samumic) [reply src/components/button/button.types.ts:17]: as disscussed

File: src/components/button/button.types.ts (hunk 4 above)
Comment (samumic) [reply src/components/button/button.types.ts:19]: i like that convention

File: src/components/button/button.tsx [hunk 9]
```diff
@@ -1,15 +1,25 @@
...
+
+    if(props.type === ButtonType.DEFAULT) base.push('bg-slate-600 hover:bg-slate-700 hover:border-slate-100 focus:border-slate-200');
```
[the remark this answers is not in the document]
Comment (samumic) [reply src/components/button/button.tsx:8]: as disscussed

**Pull request merged**